include/onnxruntime/core/platform/ort_mutex.h

```diff
@@ -101,6 +101,13 @@ std::cv_status OrtCondVar::wait_for(std::unique_lock<OrtMutex>& cond_mutex,
   return steady_clock::now() - steady_now < rel_time ? std::cv_status::no_timeout : std::cv_status::timeout;
 }
 }  // namespace onnxruntime
+#elif defined(__wasm__)
```

Comment:
Instead doing it only for wasm, let's do it more generic: Adding a new compile option just for this.

Reply:
created a cmake option "onnxruntime_ENABLE_NSYNC" and a macro "ORT_ENABLE_NSYNC" for this.